lib/carrierwave/storage/fog.rb

```diff
@@ -204,6 +204,9 @@ def authenticated_url(options = {})
                 connection.get_object_https_url(@uploader.fog_directory, path, expire_at, options)
               when 'OpenStack'
                 connection.get_object_https_url(@uploader.fog_directory, path, expire_at)
+              when 'Aliyun'
+                expire_at = expire_at - Time.now
```

Comment:
We should calculate `expire_at ` from now because inside aliyun-fog added Time.now
https://github.com/fog/fog-aliyun/blob/6f4d1716488bbcb09ba6538eb3d1cc07d2f26f4b/lib/fog/aliyun/requests/storage/get_object_http_url.rb#L25

Reply:
Could you add a test for this feature?

Reply:
What does this logic require testing?
expire_at is defined in this function and passed to another library.
I see here only the opportunity to test that the `url` method with the correct parameter was called inside the `authenticated_url` method. But this test is not of great use.